spimdisasm/common/ContextSymbols.py

```diff
@@ -65,6 +65,9 @@ class ContextSymbol:
     referenceCounter: int = 0
     "How much this symbol is referenced by something else"
 
+    referenceFunctions: Set = dataclasses.field(default_factory=set)
```

Comment:
```suggestion
    referenceFunctions: set[str] = dataclasses.field(default_factory=set)
```
Btw, what's all of that `dataclasses.field(default_factory=set)` stuffs?

Reply:
thinking a bit more about this, I wonder if it would be better to use a `ContextSymbol` reference instead of the name, since names can be a bit volatile, like determining mid-analyzis the function should be renamed because it is part of an overlay/exclusive-vram segment

Reply:
The dataclass.field stuff is the way to initialize a set in a dataclass since you can't just do `= set()` with them, otherwise you'll get a "Mutable default not allowed" error. I've changed `referenceFunctions` to a `ContextSymbol` set and verified that it works the same on Rocket.